app/js/FSPersistor.js

```diff
@@ -80,6 +92,31 @@ async function getFileSize(location, filename) {
   }
 }
 
+async function getFileMd5Hash(location, filename) {
+  const fullPath = path.join(location, filterName(filename))
+  try {
+    return await _getFileMd5HashForPath(fullPath)
+  } catch (err) {
+    throw new ReadError({
+      message: 'unable to get md5 hash from file',
+      info: { location, filename }
+    }).withCause(err)
+  }
+}
+
+async function _getFileMd5HashForPath(fullPath) {
+  return new Promise((resolve, reject) => {
+    const readStream = fs.createReadStream(fullPath)
+    const hash = crypto.createHash('md5')
+    hash.setEncoding('hex')
+    readStream.on('end', () => {
+      hash.end()
+      resolve(hash.read())
+    })
+    pipeline(readStream, hash).catch(reject)
+  })
+}
+
```

Comment:
This could be simplified a bit by returning the promise from the promisified `pipeline` instead of creating a new promise.
```
async function _getFileMd5HashForPath(fullPath) {
  const readStream = fs.createReadStream(fullPath)
  const hash = crypto.createHash('md5').setEncoding('hex')
  return pipeline(readStream, hash).then(() => {
    return hash.read()
  })
}
```